src/PSRule.Rules.Azure/rules/Azure.ContainerApp.Rule.yaml

```
@@ -98,6 +98,25 @@ spec:
       - field: properties.vnetConfiguration.infrastructureSubnetId
         hasValue: true
 
+---
+# Synopsis: Limit inbound communication for Container Apps is limited to callers within the Container Apps Environment.
+apiVersion: github.com/microsoft/PSRule/v1
+kind: Rule
+metadata:
+  name: Azure.ContainerApp.ExternalIngress
+  ref: AZR-000362
+  tags:
+    release: GA
+    ruleSet: 2023_03
+    Azure.WAF/pillar: Security
+spec:
+  with:
+  - Azure.ContainerApp.ConfigurationExternalIngress
+  - Azure.ContainerApp.WithIngress
```

Each entry in `with` is OR instead of AND which is required for this to work correctly. If you use `where` instead that will AND. Also were unlikely to need to reuse the selector for other rules.

```suggestion
  with:
  - Azure.ContainerApp.WithIngress
  where:
    value:
      $:
        configuration: AZURE_CONTAINERAPPS_RESTRICT_INGRESS
    equals: true
```